### PR TITLE
Removes delay for using reagent dispensers

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request
Removes the delay for using chem machines by unskilled marines. The current iteration doesn't do much more than delaying the start of the round. I can add more personal chem machines for personal use by the doctors if need be.

## Why It's Good For The Game
The chem machines are usually mostly used roundstart for people to make chems that they use during the rest of the round. While the delay is put to discourage this, what it mostly accomplishes is to delay the round-start since marines can just take a couple more minutes to wait until they can go groundside. I'm removing the delay in favor of not wasting more time to prep since the wait can already be very long.

Also from what I gather, the change has already been in for a couple of months and people seem to not have been discouraged.

## Changelog
:cl:
balance: removes usage delay for chem machines
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Encouraging everyone to prechem is bad. As we've seen in Minerva, extra machines and public access already encourages far more people to do it because it's easier and more accessible, and the number of marines that can do it before deployment is x4. Removing the delay is the same thing, just to a lesser degree.

Most people won't bother if they can't do it before deployment, which is honestly for the better.
If people want to go out of the way to waste their own time doing so after the fighting has actually started, then that's reducing unga effectiveness on the ground, so balances itself out.

**Comment by S1tr3p** [comment]
> Encouraging everyone to prechem is bad. As we've seen in Minerva, extra machines and public access already encourages far more people to do it because it's easier and more accessible, and the number of marines that can do it before deployment is x4. Removing the delay is the same thing, just to a lesser degree.
> 
> Most people won't bother if they can't do it before deployment, which is honestly for the better. If people want to go out of the way to waste their own time doing so after the fighting has actually started, then that's reducing unga effectiveness on the ground, so balances itself out.

Making peoples preps time longer won't discourage them from making synap oxy or any other decent premed drug. They still make it with the delay.

**Comment by Al-1ce** [comment]
>I can add more personal chem machines for personal use by the doctors if need be.

It would be a godsend.

**Comment by Derrick2Electricboogaloo** [comment]
Absolutely love this! Good work! Merge please.

**Comment by Derrick2Electricboogaloo** [comment]
> Encouraging everyone to prechem is bad. As we've seen in Minerva, extra machines and public access already encourages far more people to do it because it's easier and more accessible, and the number of marines that can do it before deployment is x4. Removing the delay is the same thing, just to a lesser degree.
> 
> Most people won't bother if they can't do it before deployment, which is honestly for the better. If people want to go out of the way to waste their own time doing so after the fighting has actually started, then that's reducing unga effectiveness on the ground, so balances itself out.

What Sitrep said, nothing will change, people will still get their meds they will just get them faster.

**Review by BraveMole — CHANGES_REQUESTED** [review]
The current situation is stupid, yes. Either we should go full locked chem, or full open chem, not just "you can have access to chem but it's pain in the ass".

But if we want to go on the open chem route, multiple things:
- There should be separate dispensers for marines in prep, they should not have to go to medbay for this. (So lock this dispenser)
- Maybe a energy-per-marine system? This would prevent marines having a whole pharmacy on them, which is just a powergamer wet dream but is making xeno life just so much harder. And then we could balance that energy number so marines get some good chems, but not all

**Comment by Walarks** [comment]
> The current situation is stupid, yes. Either we should go full locked chem, or full open chem, not just "you can have access to chem but it's pain in the ass".
> 
> But if we want to go on the open chem route, multiple things:
> 
> * There should be separate dispensers for marines in prep, they should not have to go to medbay for this. (So lock this dispenser)
> * Maybe a energy-per-marine system? This would prevent marines having a whole pharmacy on them, which is just a powergamer wet dream but is making xeno life just so much harder. And then we could balance that energy number so marines get some good chems, but not all

I could do the first part. For the second part, I'll need to look more into it since current med UIs can be interacted with by multiple people at the same time, though FOB drone is locked when someone is using it I think so I can check that.

**Comment by S1tr3p** [comment]
> The current situation is stupid, yes. Either we should go full locked chem, or full open chem, not just "you can have access to chem but it's pain in the ass".
> 
> But if we want to go on the open chem route, multiple things:
> 
> * There should be separate dispensers for marines in prep, they should not have to go to medbay for this. (So lock this dispenser)
> * Maybe a energy-per-marine system? This would prevent marines having a whole pharmacy on them, which is just a powergamer wet dream but is making xeno life just so much harder. And then we could balance that energy number so marines get some good chems, but not all

So, what about corpsmen? I can see the direction we wanna go here and its not a bad one but corpsmen often need to make a decent amount of pills with their loadout.

**Comment by BraveMole** [comment]
> So, what about corpsmen? I can see the direction we wanna go here and its not a bad one but corpsmen often need to make a decent amount of pills with their loadout.

could have more energy

**Comment by Lumipharon** [comment]
Keep in mind the medical vendor allows for infinite amounts of most basic chems, including oxycodone etc. So you can actually do a lot with just the vendor and the chem master (and the nanomed).

I think open chem is a bad idea and a really big marine buff. But if you do decide to go that way, would it not be easier to just give a light version of the medic prep vendor thing to marines in the normal prep vendors (i.e. normal marines get 20 points or whatever vs the 45~ odd medics get to spend on med stuff, and with some stuff not available)?

**Comment by Losenis** [comment]
An additional problem with making chem difficult or locked to marines is that if you're latejoining you're pretty much completely fucked since pill bottles are limited, and go out insanely fast since they're taken for loadouts as well. Sometimes they can run out in one single minute.

I've argued that if marines are too strong with chems, rather than nerf it save for the few lucky ones that joined roundstart and completely gimping the experience of everyone else, we should just buff the antichem capability of xenos. A quadrillion units of healing reagents wont do a whole lot when the marine has enough neurotox to purge, threaten death, or force hypervene wasting all those chems anyway.

To summarize, making chem more difficult or outright locking it just means "You latejoined? Go fuck yourself." and is not even a bandaid to the root problem of most xenos not having anti-chem options other than certain castes.

**Comment by bowlofcereal** [comment]
> An additional problem with making chem difficult or locked to marines is that if you're latejoining you're pretty much completely fucked since pill bottles are limited, and go out insanely fast since they're taken for loadouts as well. Sometimes they can run out in one single minute.
> 
> I've argued that if marines are too strong with chems, rather than nerf it save for the few lucky ones that joined roundstart and completely gimping the experience of everyone else, we should just buff the antichem capability of xenos. A quadrillion units of healing reagents wont do a whole lot when the marine has enough neurotox to purge, threaten death, or force hypervene wasting all those chems anyway.
> 
> To summarize, making chem more difficult or outright locking it just means "You latejoined? Go fuck yourself." and is not even a bandaid to the root problem of most xenos not having anti-chem options other than certain castes.
> 
> EDIT: Now that I am thinking about it thoroughly, the appeal of chem might be over how shit trauma kits and god forbid bandages are. Not that they're bad per se, but rather they're so absolutely inefficient in space that it's laughable if not useless at times. A fire or explosion will consume an entire stack of an item (11 bodyparts vs stacks of 10) if you don't want to risk an infection, and that's after a good while of manually applying it on you with generous use of numpad + spamclick. Meanwhile with pills that'd only take 1/16 of a slot to use with the simplicity of just two clicks to heal the entire body, so no checking for which bodypart, targetting, etc. Before any ideas of nerfing chem I'd like to see bandages and trauma kits applying to the entire body, even if their effect is nerfed in return, since otherwise there's no real effective alternative to chems other than greytiding/powergaming what the game actually lets you use involving chems.
> 
> I can tell you that as a synth, getting accidentally naded and having to heal 11 bodyparts not only with wire but also with blowtorch is such a pain in the ass that sometimes I've decided to medevac myself and cryo rather than go through it. The same extends to kits and bandages/ointment that self-examination won't tell you which bodypart is bleeding to skip on the rest. At times I've eaten bwoinks for breaking into chem roundstart, because the root problem is that the alternative to chemicals just aren't that viable other than in complement to said chems or healing dead people.

I have never seen a more true statement. The reason **nobody**, not even the medics themselves, use anykind of kit, and instead prefer to pill you, is because they are absolutely useless. The only thing they are good at is stopping bleeding on a very damaged limb so you dont have to waste isotonic injectors on them. Pills are 10000x more faster then actually closing the wound, and I feel that is a serious problem that nobody else seems to think about in these kinds of things.

Adding a delay to reagent dispensers is just annoying as fuck, and should be done away with as this PR plans to do. Make trauma, burn, and everything else kit-wise more useful or nothing will change and we will be stuck in the current endless loop of "meds too op, nerf but then realize how bad of a change it is and then revert".

**Comment by Losenis** [comment]
They're not only faster but also more efficient. Naded and all limbs have 5 damage? Either waste a full bandage stack or just 1 single pill of either bica or kelo, **or even less than a pill** if a corpsman is running tricord hypo like I used to run back in the day.

**Comment by Lumipharon** [comment]
They're mainly for corpses (as a medic), stopping bleeding as you said, or speeding healing up a tiny bit.

They would make considerably more sense if marines had plentiful access to kits, but NOT chems. I.e. they can triage themselves, stop the bleeding and avoid infections, until an actual medic can treat them.

But yeah, if everyone has access to chems, they're objectively worst most of the time.

This is kinda a bigger discussion about chem/med balance though, ultimately this PR doesn't really change a lot either way.

**Comment by usnpeepoo** [comment]
i think we should kill powergaming and preppers by removing everything related to so and only giving marines base items to work with and nothing else outside of req.

**Comment by Losenis** [comment]
You're thinking of crash

**Comment by Franky2025** [comment]
Literal cacner, fix the big picture issue of marines and chems first, supporting a development of this issue by making it easier to chem for marines is not a good idea. BraveMole suggested a great idea for marines having some sort of their own limited chem.

**Comment by Losenis** [comment]
> fix the big picture issue of marines and chems first

That same thing can be argued for getting rid of this bandaid pr that doesn't even do what it's supposed to do. Most of the delay can be bypassed by simply chaining inputs. This pr being in place if nothing else would just be justification to not touch marine chems because "well the chem machines have a big delay already".

**Comment by Franky2025** [comment]
> That same thing can be argued for getting rid of this bandaid pr that doesn't even do what it's supposed to do. Most of the delay can be bypassed by simply chaining inputs. This pr being in place if nothing else would just be justification to not touch marine chems because "well the chem machines have a big delay already".

This is literally a "lets ignore the skill system because it annoying :(". Chem access for marines is a fucking overkill always, it should not be default, all the marinemed items are fine and do their job, this removal just encourages it.

**Comment by Losenis** [comment]
> Chem access for marines is a fucking overkill always

No. Marines are mostly designed around the idea that everyone should be able to do everything because at its core TGMC isn't a AAA 2022 game with 1000s of players, some roles might end up not filled. This was the reasoning behind the pr that made marines able to install miner modules. The skill system is there to make things faster for select roles, not annoy the everloving shit of everyone who doesn't happen to have the correct icon for actual life or death stuff.

 That, and vendors are pitifully limited to the point latejoining and hoping they have chems left is wishful thinking at best. Alternative to chems like bandages, ointment, and kits are nigh useless at the time being so there's no real option other than chems, which is the actual crux of the issue that causes chem prs to be constant back and forth of buffs and nerfs to this day.

Until alternatives are made just as viable it'll be nerfing chems, marines get absolutely mercilessly stomped because they don't have anything else, chems get buffed, xenos get stomped, chems get nerfed, ad infinitum. If adding a delay to chem machines that can be easily bypassed by chaining inputs is considered a solution instead of removed for the stupid bandaid it is to hide the problem, then the problem you want solved will absolutely never be solved.

**Comment by Lumipharon** [comment]
Letting marines install miners lets them do an engineers job if they're not there/fucking about.
Marines accessing chem is so they can try avoid any reliance on medics at all cost and heal real fast during combat 24/7.

This is not equivalent. Marines already have access to basic chems from prep, and can stack bicard/kele/tricord/tram just fine. They're supposed to rely on medics to actually do their job long term. It's why pill bottles changed to smaller packets.

**Comment by Losenis** [comment]
>Marines accessing chem is so they can try avoid any reliance on medics at all cost and heal real fast during combat 24/7

Not only is this beyond hyperbole given often you have like only 3 corpsmen for 30 marines, that last part had its own pr (https://github.com/tgstation/TerraGov-Marine-Corps/pull/8561) seeking to lower the rate at which marines actually healed during combat but it got closed for some reason. And this pr has ZERO effect on actual combat healing anyway, so bringing it up is ???.

**Comment by Walarks** [comment]
Someone can complete the other requests and finish the PR if they want to. I'm too tired to do it.
